### PR TITLE
Address Compilation errors and correct `detect_current_features`.

### PR DESCRIPTION
**Pull request by MathieuDutSik**
## Motivation

Several problems were identified for the end-to-end tests and I was asked to do a separate PR about this.

## Proposal

The following corrections were made:
* Correct the compilation via the `cargo test` command.
* Make the detection of features effectively work.
* Make the compilation of the code as called from `client.rs` work for more than one feature.

## Test Plan

The CI should do it, but a separate `cargo test` also done on a computer also helps.

## Release Plan

- [x] All good!
- [ ] Need to bump the major/minor version number in the next release of the crates.
- [ ] Need to update the developer manual.
- [ ] This PR is adding or removing Cargo features.
- [ ] Release is blocked and/or tracked by other issues (see links above)

## Reviewer Checklist

- [ ] The title and the summary of the PR are short and descriptive.
- [ ] The proposed solution achieves the goals stated in the PR.
- [ ] The test plan is reproducible and provides sufficient coverage.
- [ ] The release plan is adequate.
- [ ] The commits correspond to distinct logical changes.
- [ ] The code follows the [coding guidelines](https://github.com/linera-io/linera-protocol/blob/main/CONTRIBUTING.md).
- [ ] The proposed changes look correct.
- [ ] The CI is passing.
- [x] All of the above!

### CODE REVIEW

File: linera-service/src/client.rs [hunk 1]
```diff
@@ -941,29 +941,31 @@ impl NodeService {
 #[allow(unused_mut)]
 fn detect_current_features() -> Vec<&'static str> {
     let mut features = vec![];
-    #[cfg(benchmark)]
+    // The "default" is required for the linera_indexer code.
+    features.push("default");
```
Comment (ma2bd) [inline linera-service/src/client.rs:945]: How about removing `--no-default-features` below instead?

File: linera-service/src/client.rs [hunk 2]
```diff
@@ -983,10 +985,11 @@ async fn cargo_force_build_binary(name: &'static str, package: Option<&'static s
     // Use the same features as the current environment so that we don't rebuild as often.
     let features = detect_current_features();
     if !features.is_empty() {
+        let features = features.join(",");
```
Comment (ma2bd) [inline linera-service/src/client.rs:988]: Good catch here as well.

File: linera-storage/src/scylla_db.rs [hunk 3]
```diff
@@ -59,7 +60,7 @@ impl ScyllaDbStore {
 pub type ScyllaDbStoreClient<C> = DbStoreClient<ScyllaDbClient, C>;
 
 #[cfg(any(test, feature = "test"))]
-impl ScyllaDbStoreClient<crate::TestClock> {
```
Comment (ma2bd) [inline linera-storage/src/scylla_db.rs:62]: Is this a fix or just a cosmetic choice here?

**Review by ma2bd — APPROVED** [review]
Sorry for the broken function and thanks for the fix!

File: linera-storage/src/scylla_db.rs (hunk 3 above)
Comment (MathieuDutSik) [reply linera-storage/src/scylla_db.rs:62]: It was both: a fix and cosmetic.

File: linera-service/src/client.rs (hunk 1 above)
Comment (MathieuDutSik) [reply linera-service/src/client.rs:945]: I see the situation in that respect as not completely resolved.
But the right solution is unclear to me at this point.

Comment (ma2bd) [reply linera-service/src/client.rs:945]: It seems to me that `linera-indexer` should require `rocksdb` not `default`. Let's see how the CI goes in https://github.com/linera-io/linera-protocol/pull/1030

**Pull request merged**